plutus-benchmark/ed25519-throughput/Main.hs

```diff
@@ -0,0 +1,213 @@
+   quadruples.  The DSIGN infrastructure lets us do this in a fairly generic
+   way.  However, to sign an EcdsaSecp256k1DSIGN message we can't use a raw
+   bytestring: we have to wrap it up using Crypto.Secp256k1.msg, which checks
+   that the bytestring is the right length.  This means that we have to add a
+   ByteString -> message conversion function as a parameter here.  The full
+   generality isn't need here, but let's leave it in anyway.
+-}
+mkInputs :: forall v msg .
+    (Signable v msg, DSIGNAlgorithm v, ContextDSIGN v ~ ())
+    => Integer
+    -> (ByteString -> msg)
+    -> HashFun
+    -> Inputs
+mkInputs n toMsg hash =
+    Inputs $ map mkOneInput (zip seeds1 seeds2)
+    where seedSize = 128
+          seeds = listOfSizedByteStrings (2*n) seedSize
+          seeds1 = take n seeds
+          seeds2 = drop n seeds
+          -- ^ Seeds for key generation. For some algorithms the seed has to be
+          -- a certain minimal size and there's a SeedBytesExhausted error if
+          -- it's not big enough; 128 is big enough for everything here though.
+          mkOneInput (seed1, seed2) =
+              let signKey1    = genKeyDSIGN @v $ mkSeedFromBytes seed1
+                  dataKey     = rawSerialiseVerKeyDSIGN $ deriveVerKeyDSIGN signKey1   -- public key to be checked
+                  dataKeyHash = hash dataKey :: ByteString
+                  signKey2    = genKeyDSIGN @v $ mkSeedFromBytes seed2                 -- Signing key (private)
+                  vkBytes     = rawSerialiseVerKeyDSIGN $ deriveVerKeyDSIGN signKey2   -- Verification key (public)
+                  sigBytes    = rawSerialiseSigDSIGN $ signDSIGN () (toMsg dataKeyHash) signKey2
+              in (toBuiltin vkBytes, toBuiltin sigBytes, toBuiltin dataKeyHash, toBuiltin dataKey)
+
+mkInputsAsData :: Integer -> HashFun -> BuiltinData
+mkInputsAsData n hash = Tx.dataToBuiltinData $ toData (mkInputs @Ed25519DSIGN n id hash)
+
+-- Check conditions (1) and (2) mentioned above.  We check these for all of the
+-- inputs and return True if everything succeeds and False if there's at least
+-- one failure.
+{-# INLINEABLE verifyInputs #-}
+verifyInputs :: BuiltinHashFun -> BuiltinData -> Bool
+verifyInputs hash d =
+    case Tx.fromBuiltinData d of
+      Nothing              -> Tx.error ()
+      Just (Inputs inputs) -> ver inputs True
+          where ver [] acc     = acc
+                ver (i:is) acc = ver is (acc && checkInput i)
+                checkInput (vk, sg, dkhash, dk) =
+                    let dkhash' = hash dk
+                    in dkhash == dkhash' && Tx.verifyEd25519Signature vk dkhash sg
+
+-- Create the input data, convert it to BuiltinData, and apply the verification
+-- script to that.
+mkSigCheckScript :: Integer -> UProg
+mkSigCheckScript n =
+    Tx.getPlc $ $$(Tx.compile [|| verifyInputs Tx.sha2_256 ||]) `Tx.applyCode` Tx.liftCode (mkInputsAsData n Hash.sha2_256)
+
+
+-- Protocol parameters (November 2022)
```

Comment:
Maybe we should put this somewhere common in case we want to compare other stuff to it?

Reply:
I'm not sure what you mean.  Put what somewhere common?

Reply:
They don't have to be super up-to-date, it's fine to have some approximately-right numbers for things like this, I think.

Reply:
I'll make a ticket for that.  I'm not sure where else we might want these and working that out's not very relevant to this PR.

Reply:
I moved those to near the top of the file so that they're easier to find.